storage/tests/system.py

```diff
@@ -990,3 +990,131 @@ def test_access_to_public_bucket(self):
         blob, = bucket.list_blobs(max_results=1)
         with tempfile.TemporaryFile() as stream:
             blob.download_to_file(stream)
+
+
+class TestKMSIntegration(TestStorageFiles):
+
+    FILENAMES = (
+        'file01.txt',
+    )
+
+    KEYRING_NAME = 'gcs-test'
+    KEY_NAME = 'gcs-test'
+    ALT_KEY_NAME = 'gcs-test-alternate'
+
+    def _kms_key_name(self, key_name=None):
+        if key_name is None:
+            key_name = self.KEY_NAME
+
+        return (
+            "projects/{}/"
+            "locations/{}/"
+            "keyRings/{}/"
+            "cryptoKeys/{}"
+        ).format(
+            Config.CLIENT.project,
+            'global',  # will be 'self.bucket.location' after launch
+            self.KEYRING_NAME,
+            key_name,
+        )
+
```

Comment:
This may be a no-op: Could you add a test to verify kms_key_name is visible during a object list?